src/Microsoft.AspNetCore.Razor.Tools/Internal/PackageOnlyResolveTagHelpersRunCommand.cs

```diff
@@ -32,13 +33,16 @@ public PackageOnlyResolveTagHelpersRunCommand(AssemblyLoadContext loadContext)
         public static bool TryPackageOnlyTagHelperResolution(
             CommandArgument assemblyNamesArgument,
             CommandOption protocolOption,
+            CommandOption buildBasePathOption,
             Project project,
             NuGetFramework framework,
             out int exitCode)
         {
             exitCode = 0;
 
-            if (framework.IsDesktop())
+            var buildBasePath = buildBasePathOption.Value() ?? "bin";
+
+            if (framework.IsDesktop() || Directory.Exists(buildBasePath))
```

Comment:
Is this check trying to determine if the project has been built? Seems like checking for the existence of the assembly file is the more thorough way to go as solution clean still leave a bin directory.